frontend/packages/dev-console/src/components/topology/componentFactory.ts

```diff
@@ -76,12 +78,21 @@ class ComponentFactory {
             ),
           );
         case TYPE_KNATIVE_SERVICE:
-          return withSelection(false, true)(
-            withContextMenu(
-              nodeContextMenu,
-              document.getElementById('modal-container'),
-              'odc-topology-context-menu',
-            )(KnativeService),
+          return withDndDrop<
+            any,
+            any,
+            { droppable?: boolean; hover?: boolean; canDrop?: boolean },
+            NodeProps
+          >(graphEventSourceDropTargetSpec)(
+            withEditReviewAccess()(
```

Comment:
I am able to do drag and drop even if I have no access to updates. 

In the case of the sink, it is updating the resource but in other cases on creating the connection, it sends the patch request to the resource. So, you need to update the ```withEditReviewAccess``` or create new.

Reply:
Thanks @vikram-raj , updated

Reply:
Now it is working as expected.